File: Exercices/Séance 1 - C# vs Java-20231230/ActorsDB/src/be/ipl/domaine/Person.java
package be.ipl.domaine;

import java.io.Serializable;
import java.text.SimpleDateFormat;
import java.util.Calendar;

public class Person implements Serializable {

	private static final long serialVersionUID = 1L;

	private final String name;
	private final String firstname;
	private final Calendar birthDate;
	
	public String getName() {
		return name;
	}
	
	public String getFirstname() {
		return firstname;
	}

	public String getBirthDate() {
		SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
		dateFormat.setTimeZone(birthDate.getTimeZone());
		return dateFormat.format(birthDate.getTime());
	}

	
	public Person(String name, String firstname, Calendar birthDate) {
		this.name = name;
		this.firstname = firstname;
		this.birthDate = birthDate;
	}

	@Override
	public String toString() {
		SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
		dateFormat.setTimeZone(birthDate.getTimeZone());
		return "Person [name = " + name + ", firstname = " + firstname + ", birthDate =  " + getBirthDate() + "]";
	}
	
	
}

File: Exercices/Séance 1 - C# vs Java-20231230/Ex1/Person.cs
namespace be.ipl.domaine;

[Serializable]
    internal class Person
    {

	private readonly string _name;
	private readonly string _firstname;
	private readonly DateTime _birthDate;


	// virtual allow child classes to redefine -> default behavior in Java (always virtual in Java)
    public virtual string Name { get { return _name; } }

	public string Firstname { get { return _firstname; } }


	public string BirthDate { get { return _birthDate.ToString(); } }
	


	public Person(string name, string firstname, DateTime birthDate)
	{
		this._name = name;
		this._firstname = firstname;
		this._birthDate = birthDate;
	}


	public override string ToString()
	{
		return "Person [name = " + _name + ", firstname = " + _firstname + ", birthDate =  " + BirthDate + "]";
	}


}
